core/ecschema-editing/src/test/Merging/KindOfQuantityMerger.test.ts

```diff
@@ -0,0 +1,381 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { KindOfQuantity, Schema, SchemaContext } from "@itwin/ecschema-metadata";
+import { SchemaMerger } from "../../Merging/SchemaMerger";
+import { expect } from "chai";
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+describe("KindOfQuantity merge tests", () => {
+  let targetContext: SchemaContext;
+  let sourceContext: SchemaContext;
+
+  const sourceJson = {
+    $schema: "https://dev.bentley.com/json_schemas/ec/32/ecschema",
+    name: "SourceSchema",
+    version: "1.2.3",
+    alias: "source",
+  };
+  const targetJson = {
+    $schema: "https://dev.bentley.com/json_schemas/ec/32/ecschema",
+    name: "TargetSchema",
+    version: "1.0.0",
+    alias: "target",
+  };
+  const referenceJson = {
+    $schema: "https://dev.bentley.com/json_schemas/ec/32/ecschema",
+    name: "ReferenceSchema",
+    version: "1.2.0",
+    alias: "reference",
+    items: {
+      TestUnitSystem: {
+        schemaItemType: "UnitSystem",
+      },
+      TestPhenomenon: {
+        schemaItemType: "Phenomenon",
+        definition: "Test Phenomenon",
+      },
+      TU_PER_TU: {
+        schemaItemType: "Unit",
+        unitSystem: "ReferenceSchema.TestUnitSystem",
+        phenomenon: "ReferenceSchema.TestPhenomenon",
+        definition: "TU",
```

Comment:
it may not actually matter here but I'd like to keep the units we use in testing valid.  So the definition should be TU*TU(-1) and TU should be defined as a unit.  The unit TU can have itself as its definition because in this fictional unit system TU is a base unit.  

See https://github.com/iTwin/bis-schemas/blob/master/Standard/Units.ecschema.xml for an example of real units